blog.md

```diff
@@ -0,0 +1,63 @@
+
+# Advent of Code with Cats
+
+## Day1 - Part 1
+
+### Functional Solution
+
+Even though the first part of Day1 problem can be easily solved in one line of code, we purposely break it down into multiple small functions so that we can demonstrate how to apply typeclass to the problem later. We first write a fuel() function that takes in a mass of the module, divides by three and subtracts two to find the fuel required to launch a module of a given mass. In the problem, your puzzle input is a long list of all the modules on your spacecraft. To find the fuel required for each module, you could apply the fuel() function to each module of the long list by using map function.
+ 
+This is what we’ve done by writing calculateFuels() function. Next, to sum up the list of fuels, we make use of the sum method available to Iterable like List in our case, and write a sumFuels() function. Finally, we put all of these small functions into sumOfFuel() function and solve our Part 1’s problem. 
```

Comment:
What do you think about interspersing the actual code snippets with the explanation?  

e.g.

We first write a fuel() function that takes in a mass of the module, divides by three and subtracts two to find the fuel required to launch a module of a given mass. 
```scala
def fuel(mass: Int): Int = mass / 3 - 2
```
 In the problem, your puzzle input is a long list of all the modules on your spacecraft. To find the fuel required for each module, you could apply the fuel() function to each module of the long list by using map function.  This is what we’ve done by writing calculateFuels() function. 

```scala
def calculateFuels(masses: List[Int]): List[Int] = masses.map(fuel)
```